docs/iam-usage.rst

```diff
@@ -0,0 +1,148 @@
+Using IAM For Users
+===================
+
+.. note:: This document assumes basic knowledge of Google Cloud IAM,
+   see `the docs <https://cloud.google.com/iam/docs/>`_ for details
+
+Resources that implement the IAM interface provide the following methods:
+
+Policy Editing Convenience Methods
+----------------------------------
+
+Common keyword arguments:
+
+- ``retries`` number of times to retry in the
+   even of an interrupted transaction, (not available in ``set_policy``)
+- ``version`` version number of the new policy. Not used for transactionality.
+   provided for end-user use. Left blank, will increment the version by 1.
+
+
+>>> resource.add_role(iam.user('alice@example.com'), iam.roles.OWNER.name)
```

Comment:
To add to this the `permissions` of a role will be very helpful once they are available though the API, since `testIamPermissions` deals in permissions rather than roles, in order to make a concrete role recommendation when a method call returns a 403, you will need to call 

```
valid_roles = []
missing_perms = resource.get_missing_permissions(resource.method.permissions)
for role in resource.get_grantable_roles():
    if missing_perms in role.permissions:
        valid_roles.add(role)
```

Which would get you a list of `roles` that solve your 403 problem (a potentially extremely helpful solution to opaque 403s).